datafusion/src/logical_plan/plan.rs

```diff
@@ -50,6 +52,28 @@ pub enum JoinType {
     Anti,
 }
 
+impl FromStr for JoinType {
```

Comment:
This will tie the DataFusions' API to Python datafusion API at the naming level. I am not sure we should do that:

names imo are not really datafusion's: Rust favors enums over strings. They are relevant in Python because in Python strings are prevalent. In this context, shouldn't the API for mapping a string to a join type remain specific to Pythons' implementation?

Reply:
Agreed, will change it.

Reply:
IMO it would be better if the python api also used enum (or typed string union) too, at least for autocompletion and optional type checking, but keeping it for now as strings.

Reply:
music to my hears... I ❤️ agree.